src/sentry/rules/conditions/event_frequency.py

```diff
@@ -125,3 +127,78 @@ def query_hook(self, event, start, end, environment_id):
             environment_id=environment_id,
             use_cache=True,
         )[event.group_id]
+
+
+percent_intervals = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("five minutes", timedelta(minutes=5)),
+    "10m": ("ten minutes", timedelta(minutes=10)),
+    "30m": ("30 minutes", timedelta(minutes=30)),
+    "1h": ("one hour", timedelta(minutes=60)),
+}
+
+
+class EventFrequencyPercentForm(EventFrequencyForm):
+    intervals = percent_intervals
+    interval = forms.ChoiceField(
+        choices=[
+            (key, label)
+            for key, (label, duration) in sorted(
+                percent_intervals.items(),
+                key=lambda key____label__duration: key____label__duration[1][1],
+            )
+        ]
+    )
+    value = forms.IntegerField(widget=forms.TextInput())
+
+
+class EventFrequencyPercentCondition(BaseEventFrequencyCondition):
+    label = "The issue has more errors than {value} percent of sessions in {interval}"
+
+    def __init__(self, *args, **kwargs):
+        self.intervals = percent_intervals
+        self.form_cls = EventFrequencyPercentForm
+        super().__init__(*args, **kwargs)
+
+    def query_hook(self, event, start, end, environment_id):
+        project_id = event.project_id
+        cache_key = f"r.c.spc:{project_id}-{environment_id}".replace(" ", "")
```

Comment:
Why do we need to remove spaces here? These should both just be ints, I think?

Reply:
Removing it now. Left it in carelessly from when I had the time in here